### DISCUSSION TITLE
how to use queryContent outside nuxt app ?

### DISCUSSION THREAD

**Discussion by johan-Rm** [nuxt/content#1750] [General]
I would like to generate a sitemap.xml file using the module "funkenstudio/sitemap-module-nuxt-3" and a small function calling on all the content managed by @nuxt/content

`./helpers/dynamicRoutes.ts`

```typescript
import { queryContent } from '../node_modules/@nuxt/content/dist/runtime/composables/query'

export default async () => {
    
    console.log("dynamicRoutes")
    const path = '/fr/web-pages'
    const result = await queryContent(path).find()
    const routes = result.map(r => r.path)

    return routes
}
```

I don't know how to import queryContent(), I get an error :

`ERROR  Cannot find module '#app' `

how to use (import) queryContent() outside the app ?

**Comment by farnabaz** [comment]
It is not possible to use `queryContent` outside of Nuxt Context. If you want to get the list of contents, you call query API to retrieve the list.

I didn't use the sitemap module, but as I see in the docs you can make this call using easily using `$fetch` 

```ts
export default async () => {
    const contents = await $fetch('/api/_content/query')
    const routes = result.map(r => r.path)

    return routes
}
```

**Comment by johan-Rm** [comment]
I finally opted to retrieve my JSON files this way

```typescript
const getWebPages = () => import('../content/fr/web-pages').then(r => r.default || r) 
```
